src/components/SearchInput/SearchInput.tsx

```diff
@@ -0,0 +1,130 @@
+import {
+  ChangeEvent,
+  ComponentProps,
+  FocusEvent,
+  MouseEvent,
+  ReactNode,
+  useEffect,
+  useRef,
+  useState,
+} from "react"
+import { useKeyPress } from "../../hooks/useKeyPress"
+import styles from "./SearchInput.module.css"
+
+type Omitted = "onChange" | "type"
+
+interface SearchInputProps extends Omit<ComponentProps<"input">, Omitted> {
+  "aria-label": string
+  searchIcon?: ReactNode
+  clearIcon?: ReactNode
+  debounceDelay?: number
+  value?: string
+  onSearch?: (searchTerm: string) => void
+  showSearchButton?: boolean
+}
```

Comment:
This API looks a bit different in that it takes a `value` but does not take a `onChange`. There's a slight conflict of responsibilities, who's controlling the value, this component or the parent?

It looks like this component is controlling the value, _unless_ the parent also wants to change the value, then the parent overrides the value, _unless_ there's an input event on this component, and the cycle goes on.

I think I would expect to see a `value` + `onChange` pair or just an `initialValue` and from there the component takes over that responsibility.

But of course this is mostly a philosophical matter, in practice the parent wouldn't be changing the value of `value` and this would work just fine :+1:

Reply:
Yeah, it's a little weird isn't it. I guess I was hoping to enforce the use of onSearch and originally I had added a prop of initialValue over value, except that @albatrocity pointed out that the value is technically being changed, so initialValue may not totally make sense in this case. However, I'm thinking through this further now and if we aren't using onChange, initialValue still makes sense to me, because the prop is essentially used to just pass in what the value should initially be, and the component from there handles mutating that value. I'm still pretty keen on renaming it here, but I'm curious to see what @albatrocity thinks.

Reply:
I'm on board with changing it back to `initialValue`. My concern was that if the `initialValue` prop changes, resetting the input's value to `initialValue` might be unexpected based on prop name alone. But in reality it's what I would want if triggered by a route change.